apps/frontend/src/components/TimeCardPage/SubmitCard.tsx

```diff
@@ -1,61 +1,214 @@
-    if (state === CardState.Unsubmitted) {
-      setState(CardState.InReviewEmployer);
+    if (currUser === undefined) {
+      return;
     }
-    else {
+
+    let statusEntry: StatusEntryType = undefined;
+
+    // Determine the appropriate status entry to match up with the logged in user's role
+    switch (currUser.Type) {
+      case UserTypes.Associate:
+        statusEntry = props.timesheetStatus.HoursSubmitted;
+        break;
+
+      case UserTypes.Supervisor:
+        statusEntry = props.timesheetStatus.HoursReviewed;
+        break;
+
+      case UserTypes.Admin:
+        statusEntry = props.timesheetStatus.Finalized;
+        break;
+    }
+
+    const isSubmitted = statusEntry !== undefined;
+    setSubmitted(isSubmitted);
+
+    // Set the submitted date to when the corresponding status entry was logged for the user type
+    if (isSubmitted && statusEntry.Date !== undefined) {
+      setSubmitDate(moment.unix(statusEntry.Date));
+    }
+
+    // Determine the latest status update to set the card state
+    if (props.timesheetStatus.Finalized !== undefined) {
+      setState(CardState.AdminFinalized);
+    } else if (props.timesheetStatus.HoursReviewed !== undefined) {
+      setState(CardState.InReviewAdmin);
+    } else if (props.timesheetStatus.HoursSubmitted !== undefined) {
+      setState(CardState.InReviewSupervisor);
+    } else {
       setState(CardState.Unsubmitted);
     }
-  }
+  }, [currUser, props.timesheetStatus, props.timesheetId]);
+
+  const submitAction = async () => {
+    console.log("Current user id:", currUser.UserID);
+    let statusSubmissionType: string;
+
+    // Determine the appropriate status entry to match up with the logged in user's role
+    switch (currUser.Type) {
+      case UserTypes.Associate:
+        statusSubmissionType = TimesheetStatus.HOURS_SUBMITTED;
+        break;
+
+      case UserTypes.Supervisor:
+        statusSubmissionType = TimesheetStatus.HOURS_REVIEWED;
+        break;
+
+      case UserTypes.Admin:
+        statusSubmissionType = TimesheetStatus.HOURS_SUBMITTED;
+        break;
+    }
+
+    // Update the current timesheet to be submitted by the logged-in user.
+    // The type of status can be determined on the backend by the user type
+    try {
+      const response = await ApiClient.updateTimesheet(
+        updateSchemas.TimesheetUpdateRequest.parse({
+          TimesheetID: props.timesheetId,
+          Operation: updateSchemas.TimesheetOperations.STATUS_CHANGE,
+          Payload: updateSchemas.StatusChangeRequest.parse({
+            TimesheetId: props.timesheetId,
+            AssociateId: props.associateId,
+            authorId: currUser.UserID, // TODO: Implement authorId functionality instead of dummy data
+            statusType: statusSubmissionType,
+            dateSubmitted: moment().unix(),
+          }),
+        })
+      );
+
+      // TODO: Confirm successful 2xx code responSse from API
+      props.refreshTimesheetCallback();
+
```

Comment:
We have a createToast util in TimeCardPage/utils that should be used if possible